sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -787,4 +803,82 @@ public int hashCode() {
       return Arrays.hashCode(clusteringColumns);
     }
   }
+
+  /** A TableChange to alter table and add a constraint. */
+  final class AddConstraint implements TableChange {
+    private final Constraint constraint;
+    private final String validatedTableVersion;
+
+    private AddConstraint(Constraint constraint, boolean validate) {
+      this.constraint = constraint;
+      this.validatedTableVersion = null;
```

Comment:
Hmm, how this `validatedTableVersion` can be updated? Seems it is only null?

Reply:
It will be set once we implement ALTER TABLE ... ADD CONSTRAINT. It will do a scan over the table to validate the constraint and will remember which version was checked.

I guess `validate` should be renamed to `validatedTableVersion`?

Reply:
Good catch. Updated. Missed that one on a Friday afternoon :(